server/src/lsp-server.ts

```diff
@@ -487,7 +485,13 @@ export class LspServer {
             { language: 'typescript', value: result.body.displayString }
         ];
         const tags = asTagsDocumentation(result.body.tags);
-        contents.push(result.body.documentation + (tags ? '\n\n' + tags : ''));
+        let documentation: string;
+        if (typeof result.body.documentation === 'string') {
+            documentation = result.body.documentation;
+        } else {
+            documentation = result.body.documentation?.map(p => p.text).join('') || '';
```

Comment:
According to the types this can't be undefined so:
```suggestion
            documentation = result.body.documentation.map(p => p.text).join('') || '';
```

But can you instead use the `asPlainText` helper function everywhere instead?